src/types.rs

```diff
@@ -571,40 +581,93 @@ pub trait Interface: Send + Sync + InterfaceClone {
         unimplemented!("hash_keccak256")
     }
 
-    fn amount_from_mantissa_scale(
+    fn native_amount_from_str_wasmv1(
         &self,
-        mantissa: u64,
-        scale: u32,
-    ) -> Result<u64> {
-        unimplemented!("amount_from_mantissa_scale")
+        amount: &str,
+    ) -> Result<NativeAmount> {
+        Ok(NativeAmount {
+            mandatory_mantissa: Some(100),
+            mandatory_scale: Some(0),
+        })
```

Comment:
Why did you not make this unimplemented!() like the other abis in the interface?
Here, we may call this and have unexpected behaviour if we forget to implement the interface in masse.

Reply:
hu, thanks, I wanted to implement basic stuff in the test interface… I got it wrong